### DISCUSSION TITLE
How can I prevent pinch zoom?

### DISCUSSION THREAD

**Discussion by masafumimori** [vercel/next.js#39685] [Help]
I am trying to disable pinch zoom but haven't achieved it yet.

I added below but still I can pinch zoom on my laptop.
```javascript
<meta
     name={'viewport'}
     content={'width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0'}
/>
```

I also added `touch-action: none;` style property to HTML and body element but no luck...

Does anyone know how to disable pinch zoom?

My laptop spec:
macOS Monterey version 12.5
browser: Brave Version 1.41.100 Chromium: 103.0.5060.134 (Official Build) (arm64)

**Comment by icyJoseph** [comment]
Isn't this like a WebKit thing? And in general, people have heated arguments about whether this should be done or not. 

IIRC, this is mostly fixed with JavaScript, rather than with CSS or HTML tags.

- https://www.quirksmode.org/blog/archives/2020/12/userscalableno.html
- https://stackoverflow.com/questions/37808180/disable-viewport-zooming-ios-10-safari

This answer, https://stackoverflow.com/a/52201924, seems to do the trick:

```js
document.addEventListener("gesturestart", function (e) {
	e.preventDefault();
    document.body.style.zoom = 0.99;
});

document.addEventListener("gesturechange", function (e) {
	e.preventDefault();

  document.body.style.zoom = 0.99;
});
document.addEventListener("gestureend", function (e) {
	  e.preventDefault();
    document.body.style.zoom = 1;
});
```
